src/frontend/lexer.rs

```diff
@@ -0,0 +1,259 @@
+use itertools::Itertools;
+use std::iter::Peekable;
+
+use super::token::Token;
+
+// -------------------------------------------------------------------------- //
+// Error type                                                                 //
+// -------------------------------------------------------------------------- //
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Error {
```

Comment:
Either use modules like suggested above, or rename this to be specific eg. `LexError` to ensure it doesnt get mixed up since its public?

Reply:
I should be use the modules already. See [https://github.com/NoneTheWisr/lisp-rs/pull/1#discussion_r810480998](https://github.com/NoneTheWisr/lisp-rs/pull/1#discussion_r810480998)